crypto/fipsmodule/self_check/self_check.c

```diff
@@ -1026,7 +1056,8 @@ static int boringssl_self_test_fast(void) {
 
   if (!boringssl_self_test_sha256() ||
       !boringssl_self_test_sha512() ||
-      !boringssl_self_test_hmac_sha256()) {
+      !boringssl_self_test_hmac_sha256() ||
```

Comment:
The HMAC-256 self test is called first from within the integrity test in bcm.c in https://github.com/awslabs/aws-lc/blame/08c30300a33f9000a0b11c674a8f66e4f0b21c2b/crypto/fipsmodule/bcm.c#L286, so no need to include it here.
Additional side notes:
- The sha256 test is called there as well. We can remove it, but it's ok to leave it for now.
- The hkdf test could have been embedded in this big function because it can be static, but it's ok with me as standalone.